### PR TITLE
display all contexts present in kubeconfig (#19)

### LINKED ISSUE #19: Display several clusters/contexts
We currently only display a single cluster node. The extension for VScode on the other hand displays several clusters:
image

### PR DESCRIPTION
**Pull request by adietish**
fixes #19, depends on #25, #32 
depends on https://github.com/fabric8io/kubernetes-client/issues/2215

### CODE REVIEW

**Comment by adietish** [comment]
The tree now displays contexts as it is done in vscode (only the active context - an openshift cluster - still differs):

image

I think that the labels are far to long, the current format is: `<current namespace>/<host>/<user>.`
I think that we should only display the host and decorate user and namespace like it's done ex. in the project tree:

image

@jeffmaury, @sbouchet: thoughts?

**Comment by jeffmaury** [comment]
The issue with only the host is that you will get several entries in the tree with the same value

**Comment by adietish** [comment]
> The issue with only the host is that you will get several entries in the tree with the same value

but they'd be distinguishable by the username & namespace written right to it in lighter letters?

**Comment by jeffmaury** [comment]
Yes but as it's less visible I'm afraid users will be confused

**Comment by adietish** [comment]
> Yes but as it's less visible I'm afraid users will be confused

I believe that these lengthy labels with elusively structured parts (current namespace, host, user) are more confusing than if the different components were visually comprehensible as such (bcs represented differently) .

**Comment by adietish** [comment]
> I believe that the labels are that long that you only grasp them if you carefully read them - which I think is bad UX. I believe that you should immediately grasp what cluster they concern in the first place. Then in a 2nd more careful look you should be able to distinguish the ones that concern the same cluster but different user/default namespace. 

Apparently those labels are being discussed in vscode. I extracted the suggested change for these labels to it's own issue at #42 and put it on hold until we have a decision in vscode and try to match what they'll provide.

**Comment by adietish** [comment]
This PR will be considered finished as soon as the tests are fixed (done) and the current context is labeled correctly (not done yet)

**Comment by adietish** [comment]
labelling of the active context is now corrected:

image

This PR is ready for review. 
@jeffmaury can you please review this?

File: src/main/kotlin/org/jboss/tools/intellij/kubernetes/model/util/KubeConfigContexts.kt [hunk 1]
```diff
@@ -0,0 +1,127 @@
+/*******************************************************************************
```
Comment (jeffmaury) [inline src/main/kotlin/org/jboss/tools/intellij/kubernetes/model/util/KubeConfigContexts.kt:1]: My concerns:

- I think this is a general concern utility so I would see it in intellij-common
- If the env var contains several files, my understanding is that aggregation is performed, I am wrong ? @joshuawilson  may help there

Comment (adietish) [reply src/main/kotlin/org/jboss/tools/intellij/kubernetes/model/util/KubeConfigContexts.kt:1]: > * I think this is a general concern utility so I would see it in intellij-common

This class is just a temporary solution. I want to contribute changes in kubernetes-client and think that this class wont be required any more. If there are parts which wont make it to the client library, I'll move it to intellij-common for sure.

Comment (joshuawilson) [reply src/main/kotlin/org/jboss/tools/intellij/kubernetes/model/util/KubeConfigContexts.kt:1]: I just loaded 2 different kubeconfig files in 2 different locations. They did not merge. I can switch between them with `oc login [CLUSTER_URL]` but only one is active at a time.

Comment (adietish) [reply src/main/kotlin/org/jboss/tools/intellij/kubernetes/model/util/KubeConfigContexts.kt:1]: @jeffmaury, @joshuawilson: my code that is switching to a kubeconfig that's provided in the env var is copied from kubernetes-client. There's no functional change intended compared to what the client library provides. 
I don't see any merging happening but an override instead. If there's an env var that provides a kubeconfig it'll be used, if there's none, the default kubeconfig in the home directory is what's serverd.

Comment (adietish) [reply src/main/kotlin/org/jboss/tools/intellij/kubernetes/model/util/KubeConfigContexts.kt:1]: What we also discussed is that switching the project in an OpenShift cluster doesn't work. That's intended and is adressed in #27

**Pull request merged**